sys/kern/kern_cheri_revoke.c

```diff
@@ -574,6 +579,17 @@ kern_cheri_revoke_get_shadow(struct thread *td, int flags,
 	if (!cheri_gettag(cres))
 		return (EINVAL);
 
+	vm = vmspace_acquire_ref(td->td_proc);
+	vmm = &vm->vm_map;
+	vm_map_lock_read(vmm);
+	if (!vmm->vm_cheri_revoke_quarantining) {
+		vm_map_lock_upgrade(vmm);
```

Comment:
vm_map_lock_update() can fail.

Reply:
I'll just simplify it to vm_map_lock() or now as that was probably an unwarranted micro optimization (and we'd be better off with an mmap replacement to avoid the syscall entirely if that's critical.)